packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -71,20 +87,31 @@ export class Span implements api.Span, ReadableSpan {
     parentTracer: Tracer,
     context: Context,
     spanName: string,
-    spanContext: api.SpanContext,
-    kind: api.SpanKind,
+    spanContext: SpanContext,
+    kind: SpanKind,
     parentSpanId?: string,
-    links: api.Link[] = [],
-    startTime?: api.TimeInput,
-    clock: Clock = otperformance,
+    links: Link[] = [],
+    startTime?: TimeInput,
+    _deprecatedClock?: unknown, // keeping this argument even though it is unused to ensure backwards compatibility
   ) {
-    this._clock = clock;
     this.name = spanName;
     this._spanContext = spanContext;
     this.parentSpanId = parentSpanId;
     this.kind = kind;
     this.links = links;
-    this.startTime = timeInputToHrTime(startTime ?? clock.now());
+
+    const now = Date.now();
+    this._performanceStartTime = otperformance.now();
+    this._performanceOffset = now - (this._performanceStartTime + otperformance.timeOrigin);
+
+    // if startTime is a number smaller than the start of the process
+    // assume it is a performance API timestamp and apply correction as needed
+    if (typeof startTime === 'number' && startTime < otperformance.timeOrigin) {
+      startTime += this._performanceOffset;
+    }
+
+    this.startTime = timeInputToHrTime(startTime ?? now);
```

Comment:
CC @t2t2

Reply:
Updated [codesandbox](https://codesandbox.io/s/modest-fermat-4eu09e?file=/src/otel-fixes.ts) with these changes, looks like all are working properly now 👍 

Oh it might be better to replace both references to `otperformance.timeOrigin` with getTimeOrigin from otel/core because it has additional backwards compat checks for older browsers ([IE, safari <15](https://developer.mozilla.org/en-US/docs/Web/API/Performance/timeOrigin#browser_compatibility))

https://github.com/open-telemetry/opentelemetry-js/blob/e133f3f5c3d9467c5c96df14a5fdd6c43706d450/packages/opentelemetry-core/src/common/time.ts#L47

Reply:
done. also had to export that function